### PR TITLE
Add API key header to ckan_fetch

### PR DESCRIPTION
**Pull request by sharlagelfand**
I think this is what's needed to add the API key header to `ckan_fetch()` as in @jonocarroll's comment in #122. I don't have access to a CKAN instance that needs an API key, so let me know if i'm off. thanks!

### CODE REVIEW

**Comment by jonocarroll** [comment]
Thank you! I've tested with your branch directly and it appears to work correctly 🎉 

If it helps; one thing that tripped me up with this issue was that _a_ file was indeed originally downloaded via `GET` (without the key being passed down) but it was only about 19KB (rather than the many MB I was expecting) so downstream code was broken without errors (didn't fail early). 

The file saved with the expected extension but it was corrupt when I tried to read it back in - eventually I changed the extension to `.txt` and inspected it and sure enough, it was the HTML of the 404 page.

Could I suggest an additional check (separate to this PR) that the file produced by `GET` roughly (not sure how precise is possible) matches the expected filesize from `resource_show(id)$size` in case, say, the key is invalid or the file couldn't be found (I don't think the file is guaranteed to exist, either - a preliminary call to resource_show(id) could help prevent requesting unavailable files)?

For the file I tested, I get

```
resource_show(id)$size
34717641

file.info(downloaded_file)$size
34717839
```

so maybe an exact match isn't possible, but it's hopefully clear that it's not the 404 HTML page. Maybe sizes match to within 5%?

Thanks again!

File: R/ckan_fetch.R [hunk 1]
```diff
@@ -59,15 +60,15 @@
...
+ckan_fetch <- function(x, store = "session", path = "file", format = NULL, key = get_default_key(), ...) {
   store <- match.arg(store, c("session", "disk"))
   file_fmt <- file_fmt(x)
   if (identical(file_fmt, character(0)) & is.null(format)) {
     stop("File format is not available from URL; please specify via `format` argument.")
   }
   fmt <- ifelse(identical(file_fmt, character(0)), format, file_fmt)
   fmt <- tolower(fmt)
-  res <- fetch_GET(x, store, path, format = fmt, ...)
+  res <- fetch_GET(x, store, path, format = fmt, key = get_default_key(), ...)
```
Comment (sckott) [inline R/ckan_fetch.R]: since the user can pass a `key` value into the fxn call, we need to capture that here - i.e., `key = key`

Comment (sckott) [reply R/ckan_fetch.R]: its fine if they set their key outside of the fxn call but if they pass the key to the fxn call then it won't be passed on to `fetch_GET`

Comment (sharlagelfand) [reply R/ckan_fetch.R]: fixed, thanks!

**Pull request merged**